### PR TITLE
AMIGAOS: Clean up amigaos-fs.cpp/amigaos-dialogs.cpp

### PR DESCRIPTION
**Pull request by raziel-**
amigaos-fs.cpp:
- Enhanced some of the debug output
- Renamed some variables for better readability
- Removed two debug outputs (if it works, don't bother me) :-)
- Removed lots of superfluous blank lines
- Removed two unneeded variables
- Tried to follow coding conventions

amigaos-dialogs.cpp:
- initialize pathBuffer

These only touch the AmigaOS4 platform, but i'd still be grateful for comments (especially about coding convention breaches)

Thank you

### CODE REVIEW

File: backends/fs/amigaos/amigaos-fs.cpp [hunk 1]
```diff
@@ -33,26 +33,23 @@
...
 	const char *p = str.c_str();
 
-	while (offset > 0 && (p[offset-1] == '/' || p[offset-1] == ':'))
-		offset--;
-
-	while (offset > 0 && (p[offset-1] != '/' && p[offset-1] != ':'))
-		offset--;
-
-	return p + offset;
+	while (pathOffset > 0 && (p[pathOffset-1] == '/' || p[pathOffset-1] == ':'))
```
Comment (bluegr) [inline backends/fs/amigaos/amigaos-fs.cpp:45]:
```suggestion
	while (pathOffset > 0 && (p[pathOffset - 1] == '/' || p[pathOffset - 1] == ':'))
```

File: backends/fs/amigaos/amigaos-fs.cpp [hunk 2]
```diff
@@ -33,26 +33,23 @@
...
+		pathOffset--;
+	while (pathOffset > 0 && (p[pathOffset-1] != '/' && p[pathOffset-1] != ':'))
```
Comment (bluegr) [inline backends/fs/amigaos/amigaos-fs.cpp:47]:
```suggestion
	while (pathOffset > 0 && (p[pathOffset - 1] != '/' && p[pathOffset - 1] != ':'))
```

**Comment by bluegr** [comment]
The changes look good overall, no breaches in coding conventions - just a couple of minor issues found

**Comment by raziel-** [comment]
@bluegr 

Done, thank you so much for reviewing

**Comment by raziel-** [comment]
@sev- 

Thank you for pointing out the obvious place to look...i was checking the completely wrong place

I hope i'm allowed to merge the two changes?

**Comment by bluegr** [comment]
Great work, thanks!

**Pull request merged**